make/compiler_flags

```diff
@@ -13,7 +13,7 @@ ifeq (default,$(origin CXX))
     CXX := clang++
   endif
   ifeq ($(OS),Linux)
-    CXX := g++
+    CXX := clang++
```

Comment:
A to-do here, clang-tidy uses clang, so I'm not sure how to go around having g++ here though I'd like to keep it.

Reply:
Do we really need to change this here? If we make this a part of the Jenkins process, we can set CXX = clang++ there before running make clang-tidy. Like we set flags for MPI or OPENCL before running those tests.

Reply:
Yeah idt we need it I'll remove it